### ISSUE TITLE
2.4.11 Unobscured bullet will fail the common chatboxes widgets

### ISSUE THREAD

**Issue by DavidMacDonald** [w3c/wcag#1502]
4th bullet in 2.4.11 may fail common chatboxes currently used on retail sites, unless there are no links behind it on any pages in any scroll position of the site at any point.

> Success Criterion 2.4.11 Focus Appearance (Minimum) (Level AA): For the keyboard focus indicator of each User Interface Component, all of the following are true:
4th bullet 
Unobscured: The item with focus is not entirely hidden by author-created content.

Many major retailers have a chatbox in the bottom right corner. It is an overlay, and enables clients to interact directly with a client service rep

This SC 4th bullet makes them fail. 
There may be some CSS magic to solve it, but I don't know how to create site wide non modal dialog that displaces the content below it to ensure that no content that is focusable appears below it. 

Proposal for bullet 4: 
The item with focus is not entirely hidden by author-created content ```<add>```unless it can be moved, collapsed, or dismissed by the user.```</add>```

**Comment by alastc** [comment]
Hi David,

Could you add an example or two? They don't seem to be common on UK e-commerce sites, and my knowledge of US sites goes about as far as Walmart (which doesn't have one).

**Comment by mraccess77** [comment]
These chat widgets are common and are a problem for users when collapsed as they cover content and most I've seen can't be dismissed or moved by the user (but can be collapsed if expanded).  When collapsed most of them do not entirely obscure the focused object and thus don't seem to fail this.

**Comment by DavidMacDonald** [comment]
@alastc 

They usually come up when the user is signed in. Probably the same in the UK.
Examples are Louis Vuitton, DIOR, etc...

They may be collapsed or expanded.

**Comment by DavidMacDonald** [comment]
Also, they may only be used at certain hours because there is a live person on the other end.

**Comment by alastc** [comment]
I'm trying to find a failing example... not found one yet. At larger sizes they tend to be out of the way, but even zoomed in I'm struggling.

<img src="https://user-images.githubusercontent.com/216630/98227415-37901900-1f4f-11eb-8bd4-c1ff836a1bec.png" alt="Louis Vitor homepage with chat icon covering small part of a link" style="width:300px">

<a href="https://www.virginmedia.com/shop/bundles/bigger-bundle-movies?buspart=uk_ps_acq_cable_GOOGLE_Brand_Core_Pure_Exact+(Prospects)_Brand_Core_Pure_Exact+(Prospects)_virgin+media&CMP=DFA&medium=PPC&gclid=CjwKCAiA4o79BRBvEiwAjteoYLvXb0F1s5COE-_WhOILwF5qf_mSsEukA_Waj8SMw6DVHg68lnB8SxoC1h0QAvD_BwE&gclsrc=aw.ds"><img src="https://user-images.githubusercontent.com/216630/98227834-b08f7080-1f4f-11eb-94cf-34d664a897e7.png" alt="Virgin Media homepage with live chat link on the right"></a>

I also tried several (uk based) general shops like Asda, Matalan, Target, Asos. Then Nike, Adidas. Then a couple of energy providers (very competitive here). No sign of chat windows during standard UK hours (~10am).

I've worked with a couple of clients implementing chat windows, and having it open by default wasn't even something they pushed for.

I could do with some examples, because it really doesn't seem ubiquitous, and it's hard to work out what (if any) solution would be needed.

**Comment by DavidMacDonald** [comment]
They usually are collapsed on page load. Although DIOR has had them open on page load on occasion. I've seen them as bubbles which come out into the content about 30 pixels. Here's one.

![image](https://user-images.githubusercontent.com/2365411/98449088-c4c5a000-20fe-11eb-82d0-cb35473c3dcf.png)

**Comment by DavidMacDonald** [comment]
Here's another example
![image](https://user-images.githubusercontent.com/2365411/98449186-72d14a00-20ff-11eb-940d-4b2a44c33879.png)

**Comment by DavidMacDonald** [comment]
I think this 4th bullet is going to create quite a stir once they start being cited in lawsuits in the US. I'm not saying it not worth it... but I think we need to count the costs. 

Many retailers are not watching our drafts. Maybe they should.

**Comment by mraccess77** [comment]
In @DavidMacDonald last example  it would appear it could be collapsed. - I assume that would pass if it could be collapsed and if it couldn't be collapsed then it would block for all users unless these issues crop up mostly with certain zoom levels.  The other examples are problematic for all users as well - but perhaps more so for certain zoom levels in my experience - but  again since they don't block the whole focusable element and can be scrolled I'm not sure they fail.

**Comment by DavidMacDonald** [comment]
Hi @mraccess77 
Does Level Access support the current wording? Is it just me who's concerned?

**Comment by mraccess77** [comment]
Hi @DavidMacDonald I think the premise with 2.4.11 is that it's not totally hidden - so it allows for the element to be partially hidden - this seems to allow flexibility to the author while creating a less than desirable solution for the user.  I think the assumption is that the user can then scroll the page or something to allow the whole control to be visible - but that's not a given.    So  it would appear that you could have a focus indicator with sufficient area to meet the first 3 bullets but have it covered up by other content partially and then pass the 4th bullet.  This is as opposed to saying you can't meet bullet 1-3 if something is covered up at all.  So actually the 4th bullet seems to give a nod to the idea that it's ok if it exists but is partially covered even if you can't fully expose it.   This actually seems in favor of the author which I think is the compromise that the group intended.  But perhaps I misunderstand the criteria.

**Comment by alastc** [comment]
> it allows for the element to be partially hidden

Indeed, and if people really get into the detail, you'll notice that some browsers cut-off the bottom part of an outline as you tab down a page. That shouldn't be the authors fault.

>  So actually the 4th bullet seems to give a nod to the idea that it's ok if it exists but is partially covered even if you can't fully expose it.

Sort of, we need to consider (and be explicit) whether the requirement is of the focus indicator regardless of what is around it or over it (like 2.4.7), or whether we are scoping in how the indicator is perceived including other content.

I.e. does it automatically pass if you have `a:focus {outline: 2px black solid;}` (on a light background), or do you have to consider that link in context with other things possible obscuring it.

The last bullet switches it from just considering the indicator as defined, to testing it in use.

**Comment by alastc** [comment]
Draft response (so we can put something to the group):
<hr>
It appears that the last bullet may affect some sites with hovering chat-boxes that obscure links as you tab through the page, but not all sites. 

It does appear to depend on the implementation, and it does appear to be a legitimate problem for users with disabilities. Therefore, the group is happy with the compromise the last bullet makes. It is possible to use that type of feature without failing the success criteria, and it is of benefit to users that it does pass.

**Comment by mraccess77** [comment]
I want to make sure I understand what was agreed to:
2.4.7 Focused element can be hidden but must be able to be partially or fully scrolled into view with standard keys like arrows/navigation.
2.4.11 Focused element must always be partially or fully visible when it receives focus with exceptions outside of author control
2.4.12 Focused element must always be fully visible when focused exception when out of authored controlled

**Comment by alastc** [comment]
Hi Jon,

> 2.4.11 Focused element must always be partially or fully visible when it receives focus with exceptions outside of author control

We don't have explicit exceptions for things outside author control, but the SC using "When UICs _receive_ keyboard focus" reduces the scope.

**Comment by alastc** [comment]
The group accepted the response above.
